File: backend/src/models/mot/mot_retailer.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Boolean, DateTime, Index
from sqlalchemy.dialects.postgresql import UUID
from ...config.database_config import Base


class MotRetailer(Base):
    __tablename__ = "mot_retailers"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True)
    name = Column(String(255), nullable=False)
    address_line1 = Column(String(255), nullable=False)
    address_line2 = Column(String(255))
    city = Column(String(100), nullable=False)
    county = Column(String(100))
    postcode = Column(String(20), nullable=False)
    phone = Column(String(50))
    email = Column(String(255))
    is_default = Column(Boolean, default=False)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    __table_args__ = (
        Index("ix_mot_retailers_is_default", "is_default"),
    )

File: frontend/src/models/mot/MotRetailer.ts
export interface MotRetailer {
  id: string;
  name: string;
  address_line1: string;
  address_line2?: string;
  city: string;
  county?: string;
  postcode: string;
  phone?: string;
  email?: string;
  is_default: boolean;
  is_active: boolean;
  created_at?: string;
  updated_at?: string;
}

export interface MotRetailerCreate {
  name: string;
  address_line1: string;
  address_line2?: string;
  city: string;
  county?: string;
  postcode: string;
  phone?: string;
  email?: string;
  is_default?: boolean;
}

export interface MotRetailerUpdate extends Partial<MotRetailerCreate> {
  is_active?: boolean;
}

export interface MotRetailersResponse {
  retailers: MotRetailer[];
  total: number;
}
